### PR TITLE
Feat: add api 'WithExternalPkg'.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm-go/pkg/kcl/api.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

add api 'WithExternalPkg'.

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm.go [hunk 1]
```diff
@@ -78,6 +78,8 @@ func RunFiles(paths []string, opts ...Option) (*KCLResultList, error) {
 // WithCode returns a Option which hold a kcl source code list.
 func WithCode(codes ...string) Option { return kcl.WithCode(codes...) }
 
+func WithExternalPkgs(externalPkgs ...string) Option { return kcl.WithExternalPkgs(externalPkgs...) }
```
Comment (Peefy) [inline kclvm.go:82]: More documents

File: pkg/kcl/opt.go [hunk 2]
```diff
@@ -18,7 +18,8 @@ type Option struct {
 	Err error
 }
 
-func newOption() *Option {
+// NewOption returns a new Option.
+func NewOption() *Option {
```
Comment (Peefy) [inline pkg/kcl/opt.go:22]: Why change it to public?

Comment (zong-zhe) [reply pkg/kcl/opt.go:22]: I need to use this method in `kpm` to construct an empty `Option`.

File: kclvm.go (hunk 1 above)
Comment (zong-zhe) [reply kclvm.go:82]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**